src/components/wizard/wizard.service.ts

```diff
@@ -1,19 +1,18 @@
 import { Injectable } from '@angular/core';
 import { Subject } from 'rxjs';
-import { MarqueeWizardStepComponent } from './marquee-wizard-step.component';
 
 /**
  * This service is required to provide a form of communication
  * between the marquee wizard steps and the containing marquee wizard.
- * We cannot inject the Host due to the steps being content children 
+ * We cannot inject the Host due to the steps being content children
  * rather than view children.
  */
 @Injectable()
-export class MarqueeWizardService {
-    valid$ = new Subject<MarqueeWizardValidEvent>();
+export class WizardService<TWizardStep> {
+    validChange$ = new Subject<WizardValidEvent<TWizardStep>>();
```

Comment:
I don't think this should be an `EventEmitter`? An event emitter would only be for an `@Output` whereas this is in a service?

Reply:
We've used EventEmitters in services in the past.

https://github.houston.softwaregrp.net/caf/ux-aspects-micro-focus/blob/master/src/components/side-menu/side-menu/side-menu.service.ts#L21

I don't think Subject is an appropriate type for an event.

Reply:
If you don't want it to be an event then it should probably be a complete description of the validation state of the wizard. Like a collection of observables representing the state of each step.

Reply:
It probably shouldn't have been used in that service. An `EventEmitter` is just a `Subject`:

```typescript
export interface EventEmitter<T> extends Subject<T> {
```

Comment from Angular Team:

> EventEmitter is really an Angular abstraction, and should be used pretty much only for emitting custom Events in components. Otherwise, just use Rx as if it was any other library.

See this on the proper use of EventEmitter: https://stackoverflow.com/questions/36076700/what-is-the-proper-use-of-an-eventemitter#answer-36076701

Reply:
Fair enough, so we should rework this to be stateful then?

Reply:
If we need to know the valid state of every step at any time then yes, we can update this to use a BehaviorSubject with the state of every step. If we just need to be notified when the valid state of any step changes then a Subject will suffice.

Reply:
Honestly yes to to former because essentially we are converting state in a component to an event in a service and back to state in another component. But that's beyond the scope of this issue. I thought using EventEmitter would be a quick fix to clarify a confusing system but since it's not recommended to do so, please disregard this feedback. Thanks.